homeworks/12-mobx-mars/src/state/state.ts

```diff
@@ -0,0 +1,79 @@
+import {makeAutoObservable} from "mobx"
+import {ControlsTab, DefaultRover, Photo, PhotoId, PhotoPair, Sol} from "../domain"
+import {Set, Map} from "immutable"
+import {loadRoverPhotos} from "../api"
+
+export interface State {
+    selectedTab: ControlsTab
+    selectTab: (tab: ControlsTab) => void
+
+    selectedSol: Sol
+    setCurrentSol: (sol: Sol) => void
+
+    load: () => void
+
+    selectedPhotos(): readonly PhotoPair[]
+
+    favouritePhotos(): readonly PhotoPair[]
+    addFavourite: (photoId: PhotoId) => void
+    removeFavourite: (photoId: PhotoId) => void
+}
+
+class StateImpl {
```

Comment:
I was considering to split it up somehow to make it less "monolith", but didn't think of the best way to do it, so did not do it yet.